// plugin-welcome/src/api/models/ControllersAntenatal.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersAntenatal
 */
export interface ControllersAntenatal {
    /**
     * 
     * @type {string}
     * @memberof ControllersAntenatal
     */
    added?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersAntenatal
     */
    babystatusID?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersAntenatal
     */
    patientID?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersAntenatal
     */
    userID?: number;
}

export function ControllersAntenatalFromJSON(json: any): ControllersAntenatal {
    return ControllersAntenatalFromJSONTyped(json, false);
}

export function ControllersAntenatalFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersAntenatal {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'added': !exists(json, 'added') ? undefined : json['added'],
        'babystatusID': !exists(json, 'babystatusID') ? undefined : json['babystatusID'],
        'patientID': !exists(json, 'patientID') ? undefined : json['patientID'],
        'userID': !exists(json, 'userID') ? undefined : json['userID'],
    };
}

export function ControllersAntenatalToJSON(value?: ControllersAntenatal | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'added': value.added,
        'babystatusID': value.babystatusID,
        'patientID': value.patientID,
        'userID': value.userID,
    };
}
