// plugin-welcome/src/api/models/EntAntenatal.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntAntenatalEdges,
    EntAntenatalEdgesFromJSON,
    EntAntenatalEdgesFromJSONTyped,
    EntAntenatalEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntAntenatal
 */
export interface EntAntenatal {
    /**
     * AddedTime holds the value of the "added_time" field.
     * @type {string}
     * @memberof EntAntenatal
     */
    addedTime?: string;
    /**
     * 
     * @type {EntAntenatalEdges}
     * @memberof EntAntenatal
     */
    edges?: EntAntenatalEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntAntenatal
     */
    id?: number;
}

export function EntAntenatalFromJSON(json: any): EntAntenatal {
    return EntAntenatalFromJSONTyped(json, false);
}

export function EntAntenatalFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntAntenatal {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'addedTime': !exists(json, 'added_time') ? undefined : json['added_time'],
        'edges': !exists(json, 'edges') ? undefined : EntAntenatalEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntAntenatalToJSON(value?: EntAntenatal | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'added_time': value.addedTime,
        'edges': EntAntenatalEdgesToJSON(value.edges),
        'id': value.id,
    };
}
