// plugin-welcome/src/api/models/EntAntenatalEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBabystatus,
    EntBabystatusFromJSON,
    EntBabystatusFromJSONTyped,
    EntBabystatusToJSON,
    EntPatient,
    EntPatientFromJSON,
    EntPatientFromJSONTyped,
    EntPatientToJSON,
    EntUser,
    EntUserFromJSON,
    EntUserFromJSONTyped,
    EntUserToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntAntenatalEdges
 */
export interface EntAntenatalEdges {
    /**
     * 
     * @type {EntBabystatus}
     * @memberof EntAntenatalEdges
     */
    babystatus?: EntBabystatus;
    /**
     * 
     * @type {EntPatient}
     * @memberof EntAntenatalEdges
     */
    patient?: EntPatient;
    /**
     * 
     * @type {EntUser}
     * @memberof EntAntenatalEdges
     */
    user?: EntUser;
}

export function EntAntenatalEdgesFromJSON(json: any): EntAntenatalEdges {
    return EntAntenatalEdgesFromJSONTyped(json, false);
}

export function EntAntenatalEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntAntenatalEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'babystatus': !exists(json, 'Babystatus') ? undefined : EntBabystatusFromJSON(json['Babystatus']),
        'patient': !exists(json, 'Patient') ? undefined : EntPatientFromJSON(json['Patient']),
        'user': !exists(json, 'User') ? undefined : EntUserFromJSON(json['User']),
    };
}

export function EntAntenatalEdgesToJSON(value?: EntAntenatalEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'babystatus': EntBabystatusToJSON(value.babystatus),
        'patient': EntPatientToJSON(value.patient),
        'user': EntUserToJSON(value.user),
    };
}
