// plugin-welcome/src/api/models/EntBabystatus.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntBabystatusEdges,
    EntBabystatusEdgesFromJSON,
    EntBabystatusEdgesFromJSONTyped,
    EntBabystatusEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntBabystatus
 */
export interface EntBabystatus {
    /**
     * BabystatusName holds the value of the "babystatus_name" field.
     * @type {string}
     * @memberof EntBabystatus
     */
    babystatusName?: string;
    /**
     * 
     * @type {EntBabystatusEdges}
     * @memberof EntBabystatus
     */
    edges?: EntBabystatusEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntBabystatus
     */
    id?: number;
}

export function EntBabystatusFromJSON(json: any): EntBabystatus {
    return EntBabystatusFromJSONTyped(json, false);
}

export function EntBabystatusFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntBabystatus {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'babystatusName': !exists(json, 'babystatus_name') ? undefined : json['babystatus_name'],
        'edges': !exists(json, 'edges') ? undefined : EntBabystatusEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntBabystatusToJSON(value?: EntBabystatus | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'babystatus_name': value.babystatusName,
        'edges': EntBabystatusEdgesToJSON(value.edges),
        'id': value.id,
    };
}
