// plugin-welcome/src/api/models/EntPatientEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntAntenatal,
    EntAntenatalFromJSON,
    EntAntenatalFromJSONTyped,
    EntAntenatalToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPatientEdges
 */
export interface EntPatientEdges {
    /**
     * Antenatals holds the value of the antenatals edge.
     * @type {Array<EntAntenatal>}
     * @memberof EntPatientEdges
     */
    antenatals?: Array<EntAntenatal>;
}

export function EntPatientEdgesFromJSON(json: any): EntPatientEdges {
    return EntPatientEdgesFromJSONTyped(json, false);
}

export function EntPatientEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPatientEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'antenatals': !exists(json, 'antenatals') ? undefined : ((json['antenatals'] as Array<any>).map(EntAntenatalFromJSON)),
    };
}

export function EntPatientEdgesToJSON(value?: EntPatientEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'antenatals': value.antenatals === undefined ? undefined : ((value.antenatals as Array<any>).map(EntAntenatalToJSON)),
    };
}
